### PR TITLE
Update fetch-more-data.md

### PR DESCRIPTION
**Pull request by AsafMozes**
Added incremental updates mode

### CODE REVIEW

**Comment by PRMerger19** [comment]
@AsafMozes : Thanks for your contribution! The author(s) have been notified to review your proposed change.

File: powerbi-docs/developer/visuals/fetch-more-data.md [hunk 1]
```diff
@@ -42,6 +42,12 @@ New segments are appended to the existing `dataview` and provided to the visual
 
 ## Usage in the Power BI visual
 
+### Using segments aggregation mode (default)
+With this mode, the dataviews provided to the visual contain the accumulated data for all the previous fetchMoreData requests.
+Therefore, dataviews sizes are expected to grow on each update.
```
Comment (AviSander) [inline powerbi-docs/developer/visuals/fetch-more-data.md:47]: dataview

Comment (AviSander) [reply powerbi-docs/developer/visuals/fetch-more-data.md:47]: grow WITH each update

File: powerbi-docs/developer/visuals/fetch-more-data.md [hunk 2]
```diff
@@ -94,3 +100,66 @@ As a response to calling the `this.host.fetchMoreData` method, Power BI calls th
 
 > [!NOTE]
 > To avoid client memory constraints, Power BI currently limits the fetched data total to 100 MB. You can see that the limit has been reached when fetchMoreData() returns `false`.
+
+### Using incremental updates mode
+With this mode, the dataviews provided to the visual contain just incremental data.
+Therefore, dataviews sizes are to be limited according to the defined window size.
```
Comment (AviSander) [inline powerbi-docs/developer/visuals/fetch-more-data.md:106]: dataview

File: powerbi-docs/developer/visuals/fetch-more-data.md [hunk 3]
```diff
@@ -94,3 +100,66 @@ As a response to calling the `this.host.fetchMoreData` method, Power BI calls th
...
+    }
+}
+```
+
+As a response to calling the `this.host.fetchMoreData` method, Power BI calls the `update` method of the visual with a new segment of data.
+
+> [!NOTE]
+> Altough the dataviews data between the different update is mostly exclusive, there will be some overlap between subsequent dataviews.
+> For table and categorical data mapping, it is expected the the first N dataview rows will contain data copied from the previous dataview.
+> N can be determined by: (dataView.table['lastMergeIndex'] === undefined) ? 0 : dataView.table['lastMergeIndex'] + 1
```
Comment (AviSander) [inline powerbi-docs/developer/visuals/fetch-more-data.md:164]: i think an example dataview is in place here to demonstrate what should be expected

File: powerbi-docs/developer/visuals/fetch-more-data.md [hunk 4]
```diff
@@ -94,3 +100,66 @@ As a response to calling the `this.host.fetchMoreData` method, Power BI calls th
...
+
```
Comment (AviSander) [inline powerbi-docs/developer/visuals/fetch-more-data.md:165]: as discussed, let's add API limitations\boundaries: 

- reminder 30K rows is maximum 

- max 1M rows enforced by the server

- max 100MB dataview size enforced by the client

**Review by AviSander — CHANGES_REQUESTED** [review]
see comments

**Comment by PRMerger9** [comment]
@AsafMozes : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by PRMerger9** [comment]
@AsafMozes : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by PRMerger10** [comment]
@AsafMozes : Thanks for your contribution! The author(s) have been notified to review your proposed change.

File: powerbi-docs/developer/visuals/fetch-more-data.md [hunk 5]
```diff
@@ -94,3 +106,75 @@ As a response to calling the `this.host.fetchMoreData` method, Power BI calls th
...
+            // for example, when the 100 MB limit has been reached
+        }
+    }
+}
+```
+
+As a response to calling the `this.host.fetchMoreData` method, Power BI calls the `update` method of the visual with a new segment of data.
+
+> [!NOTE]
+> Altough the dataviews data between the different update is mostly exclusive, there will be some overlap between subsequent dataviews.
```
Comment (AviSander) [inline powerbi-docs/developer/visuals/fetch-more-data.md:177]: Although

**Review by AviSander — APPROVED** [review]
one last typo

**Comment by PRMerger18** [comment]
@AsafMozes : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by PRMerger13** [comment]
@ktoliver : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by ktoliver** [comment]
@AviSander Thanks very much for your review of the PR. In the public repo, PRs need to be approved by a member of the content team or the PM for the service.

@KesemSharabi Could you review the PR? Thanks.

**Comment by KesemSharabi** [comment]
#sign-off

**Comment by PRMerger15** [comment]
@AsafMozes : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by KesemSharabi** [comment]
#hold-off

Please don't publish this article until 10 November.

Thanks,

Kesem

**Comment by ktoliver** [comment]
> #hold-off
> 
> Please don't publish this article until 10 November.
> 
> Thanks,
> 
> Kesem

Hi @KesemSharabi - We don't have a schedule-based publishing pipeline available, but we will leave this in the queue until you sign off again. Note that this PR is in the public repo and visible to all GitHub users.

-PR review team

**Comment by PRMerger6** [comment]
@ktoliver : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by KesemSharabi** [comment]
Hi @AsafMozes, @AviSander, @ktoliver and @maggiesMSFT,

I understood from Ranin yesterday that she wants this article published on 10 November, just before her blog post goes live. As this is a public repo and we try to keep PRs on it closed, I opened my own PR on my private repo with the content of this file (fetch-more-data.md). I'll publish my PR on 10 November and we will close this one.

My PR is:

https://github.com/MicrosoftDocs/powerbi-docs-pr/pull/4691

@maggiesMSFT - please go ahead and close this PR.

Thanks,

Kesem

**Comment by maggiesMSFT** [comment]
Thanks, @KesemSharabi , @AsafMozes , and @AviSander !

**Pull request closed without merging**